filepattern-generator-plugin/src/main.py

```diff
@@ -0,0 +1,201 @@
+import argparse, logging, os
+from curses.ascii import EM
+from queue import Empty
+import filepattern 
+from pathlib import Path
+from numpy import empty
+import pandas as pd
+from typing import Optional
+import pyarrow.feather 
+import time
+
+
+
+class Filepattern_Generator:
+    def __init__(self, 
+                inpDir:Path,
+                outDir:Path,
+                pattern:Optional[str]= None,
+                chunkSize:Optional[int]= 30,
+                groupBy: Optional[str]= None,
+                outFormat: Optional[str] = None
+                ):
+        """
+        Parameters
+        ----------
+        inpDir : Path
+            Input image collection
+        outDir : Path
+            Ouput Collection
+        pattern: : str, optional
+            Filepattern regex to parse image files
+        chunkSize: : int, optional
+            Number of images to generate collective filepattern (default value=30)
+        groupBy: : str, optional
+            Select a parameter to generate filepatterns in specific order
+        outFormat: : str, optional
+            Output Format of collective filepatterns. Only Supports (CSV and feather) file format. (default file format is CSV)
+
+        Returns
+        -------
+        A collection of generated filepatterns in either (CSV and feather) file format
+
+        """
+        self.inpDir=Path(inpDir)
+        self.outDir=Path(outDir)
+        self.pattern=pattern
+        self.chunkSize=chunkSize
+        self.groupBy=groupBy
+        self.outFormat=outFormat
+        if self.pattern is None:
+            self.fileslist = [f.name for f in self.inpDir.iterdir() if f.with_suffix('.ome.tif')]
```

Comment:
This seems to say that if you don't specify a file pattern, we only support ome.tif format?  

I would probably want to error if someone passes in a different file pattern.

Reply:
Do you want to use it for all types of images not particularly OME TIFF images then in that case I modify it since it is only implemented in the case where user don't provide pattern?

Reply:
Not sure.  That may be something you should ask Nick.  

I just notice that we have a pretty string condition that if you don't specify format, than we assume ome.tif.  

If this plugin can only take a .ome.tif file, then thats fine.